engines/ceg_prophecy_arrays.cpp

```diff
@@ -158,9 +160,11 @@ void CegProphecyArrays<Prover_T>::initialize()
 template <class Prover_T>
 void CegProphecyArrays<Prover_T>::cegar_abstract()
 {
+  aa_.do_abstraction();
+  aae_.initialize();
   // the ArrayAbstractor already abstracted the transition system on
   // construction -- only need to abstract bad
-  Term prop_term = (super::ts_.solver() == super::orig_property_.solver())
+  Term prop_term = (abs_ts_.solver() == super::orig_property_.solver())
                        ? super::orig_property_.prop()
                        : super::to_prover_solver_.transfer_term(
                            super::orig_property_.prop());
```

Comment:
minor thing I just noticed, this should probably be: `to_prover_solver_.transfer_term(super::orig_property_.prop(), BOOL)`. Basically that handles the case where the original solver was boolector and the property is a size-one bit-vector, then it will cast it to be a boolean if the current solver is not boolector.

Reply:
similar situation might happen elsewhere. What do you think about creating a separate PR for that?

Reply:
Yep I'll take care of it.